src/fivetran-destination/src/destination.rs

```diff
@@ -241,11 +275,22 @@ impl TryFrom<&crate::fivetran_sdk::Column> for ColumnMetadata {
 
     fn try_from(value: &crate::fivetran_sdk::Column) -> Result<Self, Self::Error> {
         let escaped_name = escape::escape_identifier(&value.name);
+
         let mut ty: Cow<'static, str> = utils::to_materialize_type(value.r#type())?.into();
-        if let Some(d) = &value.decimal {
-            ty.to_mut()
-                .push_str(&format!("({}, {})", d.precision, d.scale));
-        }
+        let params = value.params.as_ref().and_then(|p| p.params.as_ref());
+
+        match (value.r#type(), params) {
+            (DataType::Decimal, Some(Params::Decimal(DecimalParams { precision, scale }))) => {
+                ty.to_mut().push_str(&format!("({}, {})", precision, scale));
+            }
+            (other, Some(Params::Decimal(DecimalParams { .. }))) => {
+                let msg = format!("found decimal params for {other:?} data type!");
```

Comment:
nit: Is the exclamation mark necessary?

Reply:
It's not, I like including it because it's seriously unexpected, but happy to remove if you'd prefer